main.py

```diff
@@ -158,24 +158,55 @@ async def bot_info(interaction: discord.Interaction):
 @tree.command()
 async def battle(interaction: discord.Interaction):
     '''Have a battle with a random bot!'''
+    
     async with interaction.channel.typing():
         await asyncio.sleep(1)
+        
+        
+    curr_season = server_config['season']
+
+    #Get leaderboard info
+    response = await rocketbot_client.query_leaderboard(curr_season)
+    response = json.loads(response['payload'])['records']
+    
+    rand_player = random.choices(response)['username']
     
-    event = random.choices(
-        population=[
-            "The bot dodged your attack. <:bot:917467970182189056>",
-            "You thought you hit the bot, but its health goes back up due to network lag.",
-            "You destroyed the bot!\nIt drops one coin. <:coin:910247623787700264>",
-            "The bot vanishes. An error pops up: `CLIENT DISCONNECTED` <:alertbad:910249086299557888>"
-        ],
-        weights=[
-            90,
-            5,
-            10,
-            1,
-        ],
-        k=1,
-    )[0]
+    events = {
+        "The bot dodged your attack. <:bot:917467970182189056>"
+        : 70,
+        "You destroyed the bot! It drops a single coin. <:coin:910247623787700264>"
+        : 10,
+        "The bot *expertly* dodged your attack. <:bot:917467970182189056>"
+        : 5,
+        "You thought you hit the bot, but its health returns to full due to network lag. 📶"
+        : 5,
+        "You destroyed the bot! It drops a some coins and a crate. <:coin:910247623787700264> <:coin:910247623787700264> <:coin:910247623787700264> 📦. But " + rand_player + " comes out of nowhere and steals it."
```

Comment:
If this is the only instance we use a random player we shouldn't be querying the leaderboard every time